docs/pages/guides/prebuild.md

```diff
@@ -0,0 +1,168 @@
+
+## Templates
+
+You can customize how the native folders are generated while remaining in the managed workflow by building [Expo Config Plugins][config-plugins]. Many config plugins already exist for lots of modifications. You can see the [list][config-plugins-repo] for more information.
+
+Prebuild generates template files before modifying them with Expo Config Plugins. The template files are versioned and come from the npm package [`expo-template-bare-minimum`][template]. You can change the template used by passing `--template /path/to/template.tgz` to the `npx expo prebuild` command. This is not recommended as the base modifiers in `@expo/prebuild-config` make some undocumented assumptions about the template files.
+
+## Side-Effects
+
+As of SDK 46, `npx expo prebuild` has some unfortunate side-effects that we plan to remove. Ideally, running `npx expo prebuild` would only generate an `ios` and `android` folder but this is currently not the case.
+
+In addition to generating the native folders, prebuild also makes the following modifications:
+
+- Generating a `metro.config.js` file.
+- Generating an `index.js` file.
+- Removing the `main` field in the `package.json`.
+- Modifying the `scripts` field in the `package.json`.
+- Modifying the `dependencies` field in the `package.json`.
+
+The convenience change to the `scripts` field is the only thing that alters how a user develops their app before/after prebuild, all other changes can be left as-is if desired.
+
+## Optionality
+
+Prebuilding is completely optional and does not disqualify a project from using other features or services offered by Expo. We created the designation _bare workflow_ to refer projects that do not use `npx expo prebuild`.
+
+Everything offered by Expo including [EAS][eas], Expo CLI, and the libraries in the Expo SDK are built to **fully support** the _bare workflow_ as this is a minimum requirement for supporting projects using `npx expo prebuild`.
+
+The main exception is the [Expo Go][expo-go] app which has partial support for the _bare workflow_, given the project in question only utilizes native features that are already available in the Expo Go app.
+
+We develop new [Native Modules][native-modules] without using `npx expo prebuild` as it's currently the fastest way to do so. However, we recommend users building apps with Expo attempt to use Prebuild as much as possible for the reasons listed in the [pitch](#pitch) section.
+
+## Pitch
+
+A single native project on its own is very complicated to maintain, scale, and grow. In a cross-platform app, you have multiple native projects that you must maintain and keep them up to date. We created the _optional_ Expo Prebuild system to streamline this process. Below are a few issues we've identified with native development and some corresponding reasons we believe Expo Prebuild solves these issues.
```

Comment:
```suggestion
A single native project on its own is complicated to maintain, scale, and grow. In a cross-platform app, you have multiple native projects that you must maintain and keep up to date for the latest operating system releases and to avoid falling too far behind in any third party dependencies. We created the _optional_ Expo Prebuild system to streamline this process. Below are a few issues we've identified with native development in the context of React Native and some corresponding reasons we believe Expo Prebuild solves these issues.
```